cogeo_mosaic/backends/base.py

```diff
@@ -0,0 +1,61 @@
+"""cogeo_mosaic.backend.base: base Backend class."""
+
+import abc
+from contextlib import AbstractContextManager
+from typing import Dict, Optional, Tuple
+
+from cogeo_mosaic import version as mosaic_version
+from cogeo_mosaic.backends.utils import get_hash
+from cogeo_mosaic.model import MosaicJSON
+
+
+class BaseBackend(AbstractContextManager):
+    mosaic_def: MosaicJSON
+
+    @abc.abstractmethod
+    def __init__(self, *args, **kwargs):
+        """Connect to backend"""
+
+    def __enter__(self):
+        """Support using with Context Managers"""
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Support using with Context Managers"""
+
+    def metadata(self) -> Dict:
+        """Retrieve Mosaic metadata
+
+        Returns
+        -------
+        MosaicJSON as dict without `tiles` key.
+        """
+        return {k: v for k, v in dict(self.mosaic_def).items() if k != "tiles"}
+
+    @abc.abstractmethod
+    def tile(self, x: int, y: int, z: int) -> Tuple[str]:
+        """Retrieve assets for tile."""
+
+    @abc.abstractmethod
+    def point(self, lng: float, lat: float) -> Tuple[str]:
+        """Retrieve assets for point."""
+
+    @abc.abstractmethod
+    def read_mosaic(self, *args, **kwargs) -> MosaicJSON:
```

Comment:
not sure if `read` will make more sense here 
cc @geospatial-jeff @kylebarron

Reply:
I thought that read would be too confusing because you're not actually reading any data. A `read` method on a backend class would generally make me think I'm reading data (assets), not just the metadata

Reply:
I think `read` is just saying the backend is accessing its data which is, in this case, the mosaic definition.